### PR TITLE
TA158782: Add file attachments component

### PR DESCRIPTION
**Pull request by brissons**
## Relevant Rally Links

- [TA158782](https://rally1.rallydev.com/#/289692574792/custom/355050439968?detail=%2Ftask%2F612885593299&fdp=true): Investigate and implement d2l-attachment component

## Description

This task was to investigate if we could use the `d2l-attachment` component in the way dictated by our designs. The result of this investigation was an implementation of the attachment component for files.

**Note**: This has only been tested with files from manage files. The `href` would change, so this will need to be checked with content service files. ~~I think I'm going to leave this PR in review until then.~~ This will be merged and any changes needed for content service will be added later.

## Goal/Solution

Use the `d2l-attachment` component with the non-embedded view for files.

## Video/Screenshots

image

### CODE REVIEW

**Review by ianlaidlaw — APPROVED** [review]

File: components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-media-file-detail.js [hunk 1]
```diff
@@ -129,19 +129,24 @@ class ContentMediaFileDetail extends SkeletonMixin(ErrorHandlingMixin(LocalizeAc
...
-			name: 'TODO',
-			url: mediaFileEntity.fileLocationHref
+			name: this._getEntityName(mediaFileEntity),
+			url: mediaFileEntity.fileLocationHref,
+			type: 'Document'
+		};
+
+		//this is only for enabling the link; actual permission checking is not handled here
+		const hasPermission = {
+			canAccess: function() { return true; }
```
Comment (ianlaidlaw) [inline components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-media-file-detail.js:139]: super duper small nit, i feel like generally we would use arrow functions for these throwaway functions but its totally up to you

Comment (ianlaidlaw) [reply components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-media-file-detail.js:139]:
```
canAccess: () => true;
```
might work

**Review by coffeexcode — DISMISSED** [review]

**Review by Kiefer-Co-D2L — DISMISSED** [review]

File: components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-media-file-detail.js (hunk 1 above)
Comment (brissons) [reply components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-media-file-detail.js:139]: Yeah I like this better.

**Review by coffeexcode — APPROVED** [review]
LGTM

**Pull request merged**